siad/api.go

```diff
@@ -191,6 +192,30 @@ func (d *daemon) downloadHandler(w http.ResponseWriter, req *http.Request) {
 	*/
 }
 
+func (d *daemon) updateHandler(w http.ResponseWriter, req *http.Request) {
+	enc := json.NewEncoder(w)
+
+	switch req.FormValue("action") {
+	case "check":
+		available, err := d.checkForUpdate()
+		enc.Encode(struct {
+			Available bool
+			Error     string
+		}{available, err.Error()})
+
+	case "apply":
+		applied, err := d.applyUpdate()
+		enc.Encode(struct {
+			Applied bool
+			Error   string
+		}{applied, err.Error()})
```

Comment:
going to convert the rest of the API to this style (JSON only). I thought it'd be a pain at first, but the use of `NewEncoder` and struct literals makes things fairly clean.